File: 2022/day-02/python/part2_sol.py
f = open("../input.txt")
data = f.read()
strategy = [i.split(" ") for i in data.split("\n")]
# print(strategy)

score = 0


for s in strategy:
    result = (ord(s[1]) - 87)%3
    opponent = (ord(s[0]) - 64)%3
    player = (result + opponent)%3
    score += (ord(s[1]) - 88)*3 + player + 1

print(score)

File: 2022/day-02/rust/part2_sol.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("../input.txt").expect("Couldn't read from file");
    let mut score = 0;
    for line in data.lines() {
        let s: Vec<&str> = line.split(" ").collect();
        if s[0] == "A" {
            if s[1] == "Y" {
                score += 3 + 1;
            } else if s[1] == "Z" {
                score += 6 + 2;
            } else {
                score += 0 + 3;
            }
        } else if s[0] == "B" {
            if s[1] == "Y" {
                score += 3 + 2;
            } else if s[1] == "Z" {
                score += 6 + 3;
            } else {
                score += 0 + 1;
            }
        } else {
            if s[1] == "Y" {
                score += 3 + 3;
            } else if s[1] == "Z" {
                score += 6 + 1;
            } else {
                score += 0 + 2;
            }
        }
    }

    println!("{}", score);
}